EIPS/eip-FOCIL.md

```diff
@@ -16,28 +16,73 @@ Implement a robust mechanism to preserve Ethereum’s censorship resistance and
+
+## High-level Overview
+
+### Roles And Participants
+
+This section outlines the workflow of FOCIL, detailing the roles and responsibilities of various participants, including IL committee members, nodes, proposers, and attesters.
+
+##### IL Committee Members
+
+- **`Slot N`, `t=0 to 8s`**:
+IL committee members construct their local ILs and broadcast them over the P2P network after processing the block for `slot N` and confirming it as the head. If no block is received by `t=7s`, they should run `get_head` and build and release their local ILs based on their node’s canonical head.
+
+  By default, local ILs are built by selecting raw transactions from the public mempool, ordered by priority fees, up to the local IL’s maximum size in bits (e.g., 8 KB per local IL). Additional local rules can be optionally applied to maximize censorship resistance, such as prioritizing valid transactions that have been pending in the mempool the longest.
+
+##### Nodes
+
+- **`Slot N`, `t=0 to 9s`**:
+Nodes receive local ILs from the P2P network and only forward and cache those that pass the CL P2P validation rules.
+
+- **`Slot N`, `t=9s`**:, The local IL freeze deadline:
+Nodes freeze their local ILs view, stop forwarding and caching new local ILs.
+
+##### Proposer
+- **`Slot N`, `t=0 to 11s`**: The proposer receives local ILs from the P2P network, forwarding and caching those that pass the CL P2P validation rules. Optionally, an RPC endpoint can be added to allow the proposer to request the missing local ILs from its peers (e.g., by committee index at `t=10s`).
+
+- **`Slot N`, `t=11s`**:
+The proposer freezes its view of local ILs and asks the EL to update its execution payload by adding transactions from its view (the exact timings will be defined after running some tests/benchmarks).
+
+- **`Slot N+1`, `t=0s`**:
+The proposer broadcasts its block with the up-to-date execution payload satisfying IL transactions over the P2P network.
+
+##### Attesters
+- **`Slot N+1`, `t=0 to 4s`**:
+Attesters monitor the P2P network for the proposer’s block. Upon detecting it, they verify whether all transactions from their cached local ILs are included in the proposer’s execution payload. The `Valid` function, based on the frozen view of the local ILs from `t=9s` in the previous slot, checks if the execution payload satisfies IL validity conditions. This is done either by confirming that all transactions are present or by determining if any missing transactions are invalid when appended to the end of the payload. In such cases, attesters use the EL to perform nonce and balance checks to validate the missing transactions and check whether there is enough space in the block to include the transaction(s).
+
+#### CL P2P Validation Rules
```

Comment:
I would add a bit of explanations here, for example state that the IL is allowed to contain anything, as there is no EL-side verification.

Reply:
Added: "When nodes receive local ILs from the P2P network, they perform a series of validation checks before forwarding or caching them. At this stage, there is no EL verification of the transactions within local ILs. This means that ILs are allowed to contain any transactions—valid or invalid—since nodes do not perform EL-side checks. This design choice is intended to avoid additional computational overhead."